SigningService/Agents/SignAgent.cs

```diff
@@ -45,7 +46,7 @@ private void SignPackagePart(PackagePart packagePart)
         {
             foreach (var signer in _signerRepository)
             {
-                signer.TrySign(packagePart);
+                signer.TrySignAsync(packagePart.GetStream()).Wait();
```

Comment:
we should be using the async model here. Otherwise, we leave ourselves open to deadlocks, particularly in ASP.NET scenarios.

private Package Sign(Package package) --> private Task<Package> Sign(Package package)
private void SignPackagePart(PackagePart packagePart) --> private Task SignPackagePart(PackagePart packagePart)